src/general-components/Tool/StepComponent/StepComponent.tsx

```diff
@@ -244,12 +295,20 @@ class StepComponent extends Component<StepComponentProps, StepComponentState> {
             if (step.values && Object.keys(step.values).length !== 0) {
                 progress++;
                 step.ref.current?.setValues(step.values);
-                step.ref.current?.setDisabled(true);
                 step.ref.current?.rebuildValues(step.values);
+
+                if (
+                    progress < this.state.steps.length
+                    && Object.keys(this.state.steps[progress].values).length > 0
+                ) {
+                    step.ref.current?.setDisabled(true);
```

Comment:
It should get documented somewhere, which surcomstances result in a step getting locked.